flink-core/src/test/java/org/apache/flink/api/java/ClosureCleanerTest.java

```diff
@@ -133,6 +135,27 @@ void testComplexTopLevelClassClean() throws Exception {
         assertThat(result).isEqualTo(5);
     }
 
+    @Test
+    public void testCleanNonSerializableNestedMap() throws Exception {
```

Comment:
Thanks for adding this test. Could you please add another test covering a scenario where multiple reference classes are involved?

Reply:
There are multiple classes here in the test: `ComplexMap` has `map1` field which is a `MapFunction`. This MapFunction refers to a non-serializable Object. Meanwhile `ComplexMap` also has `map4` field  which is a `LocalMap`. This LocalMap also refers to same value of `map1`. So it has 4 layers of references.